modules/core/src/main/java/org/apache/synapse/api/rest/RestRequestHandler.java

```diff
@@ -69,8 +72,14 @@ public boolean process(MessageContext synCtx) {
 
     @Override
     protected boolean dispatchToAPI(MessageContext synCtx) {
-        Collection<API> apis = synCtx.getEnvironment().getSynapseConfiguration().getAPIs(
-                ApiConstants.DEFAULT_BINDING_ENDPOINT_NAME);
+        Object apiObject = synCtx.getProperty(RESTConstants.PROCESSED_API);
+        Collection<API> apis;
+        if (apiObject != null) {
+            apis = Collections.singletonList((API) apiObject);
+        } else {
+            apis = synCtx.getEnvironment().getSynapseConfiguration().getAPIs(
+                    ApiConstants.DEFAULT_BINDING_ENDPOINT_NAME);
+        }
```

Comment:
isn't this anyway done in abstract class, here https://github.com/wso2/wso2-synapse/blob/master/modules/core/src/main/java/org/apache/synapse/api/AbstractApiHandler.java#L52 when called in line 84

Reply:
RestRequestHandler hits first before the AbstractAPIHandler Either We handle the code at AbstractAPIHandler, code get obselete due to 
           apis = synCtx.getEnvironment().getSynapseConfiguration().getAPIs(
                    ApiConstants.DEFAULT_BINDING_ENDPOINT_NAME);
Lock the SynapseConfiguration Object.